src/GraphCoarsenBuilder.cpp

```diff
@@ -110,77 +104,46 @@ void ElementMBuilder::SetBubbleBubbleBlock(int l, int j, double value)
 
 void ElementMBuilder::ResetEdgeCdofMarkers(int size)
 {
-    edge_cdof_marker_.SetSize(size);
-    edge_cdof_marker_ = -1;
-    edge_cdof_marker2_.SetSize(size);
-    edge_cdof_marker2_ = -1;
+    edge_cdof_markers_[0].resize(size, -1);
+    edge_cdof_markers_[1].resize(size, -1);
 }
 
 void ElementMBuilder::FillEdgeCdofMarkers(int face_num, const mfem::SparseMatrix& face_Agg,
                                           const mfem::SparseMatrix& Agg_cdof_edge)
 {
-    mfem::Array<int> Aggs;
     mfem::Array<int> local_Agg_edge_cdof;
-    GetTableRow(face_Agg, face_num, Aggs);
-    Agg0_ = Aggs[0];
-    GetTableRow(Agg_cdof_edge, Agg0_, local_Agg_edge_cdof);
-    for (int k = 0; k < local_Agg_edge_cdof.Size(); k++)
-    {
-        edge_cdof_marker_[local_Agg_edge_cdof[k]] = k;
-    }
-    if (Aggs.Size() == 2)
+    GetTableRow(face_Agg, face_num, Aggs_);
+    for (int a = 0; a < Aggs_.Size(); a++)
     {
-        Agg1_ = Aggs[1];
-        GetTableRow(Agg_cdof_edge, Agg1_, local_Agg_edge_cdof);
+        GetTableRow(Agg_cdof_edge, Aggs_[a], local_Agg_edge_cdof);
         for (int k = 0; k < local_Agg_edge_cdof.Size(); k++)
         {
-            edge_cdof_marker2_[local_Agg_edge_cdof[k]] = k;
+            edge_cdof_markers_[a][local_Agg_edge_cdof[k]] = k;
         }
     }
-    else
-    {
-        Agg1_ = -1;
-    }
 }
 
-void ElementMBuilder::AddTraceAcross(int row, int col, double value)
+void ElementMBuilder::AddTraceAcross(int row, int col, int agg, double value)
 {
-    mfem::DenseMatrix& CM_el_loc1(CM_el_[Agg0_]);
+    mfem::DenseMatrix& CM_el_loc(CM_el_[Aggs_[agg]]);
 
-    int id0_in_Agg0 = edge_cdof_marker_[row];
-    int id1_in_Agg0 = edge_cdof_marker_[col];
-    if (Agg1_ == -1)
-    {
-        CM_el_loc1(id0_in_Agg0, id1_in_Agg0) += value;
-    }
-    else
-    {
-        mfem::DenseMatrix& CM_el_loc2(CM_el_[Agg1_]);
-        CM_el_loc1(id0_in_Agg0, id1_in_Agg0) += value / 2.;
-        int id0_in_Agg1 = edge_cdof_marker2_[row];
-        int id1_in_Agg1 = edge_cdof_marker2_[col];
-        CM_el_loc2(id0_in_Agg1, id1_in_Agg1) += value / 2.;
-    }
+    int id0_in_agg = edge_cdof_markers_[agg][row];
+    int id1_in_agg = edge_cdof_markers_[agg][col];
+    CM_el_loc(id0_in_agg, id1_in_agg) += value;
```

Comment:
This all looks much simpler and cleaner, let me make sure I understand: now in this code you no longer divide by 2 for these local matrices at this stage, but when you actually assemble the matrix (around line 146) the `agg_weight` includes the appropriate scaling, so the final result is the same as before?